sdk/iOS/src/MSSyncTable.h

```diff
@@ -98,17 +98,17 @@
 /// MSQeury object.
 /// Before a pull is allowed to run, all pending requests on the specified table will be sent to
 /// the server. If a pending request for this table fails, the pull will be cancelled
--(void)pullWithQuery:(MSQuery *)query queryId:(NSString *)queryId completion:(MSSyncBlock)completion;
+-(void)pullWithQuery:(nonnull MSQuery *)query queryId:(nullable NSString *)queryId completion:(nullable MSSyncBlock)completion;
 
 /// Removes all records in the local cache that match the results of the specified query.
 /// If query is nil, all records in the local table will be removed.
 /// Before local data is removed, a check will be made for pending operations on this table. If
 /// any are found the purge will be cancelled and an error returned.
--(void)purgeWithQuery:(MSQuery *)query completion:(MSSyncBlock)completion;
+-(void)purgeWithQuery:(nullable MSQuery *)query completion:(nullable MSSyncBlock)completion;
```

Comment:
In the implementation, you have a check for query == nil here https://github.com/Azure/azure-mobile-services/blob/MobileApp/sdk/iOS/src/MSSyncTable.m#L70-L74

Hence the nullable annotation.